### PR TITLE
[14.0] edi: misc fix/imp

### PR DESCRIPTION
**Pull request by simahawk**
Please, see atomic commits for details.
Most important changes:

* fixed the ordering which was just random due to the override of `_search` CC @etobella 
* added many handy filters and additional menu items
* moved q.job relation from backend to exchange record (see motivation in its commit)

### CODE REVIEW

**Review by ivantodorovich — APPROVED** [review]
LG

File: edi_oca/data/job_function.xml [hunk 1]
```diff
@@ -1,26 +1,26 @@
 <odoo noupdate="1">
-    <record id="job_edi_backend_record_generate" model="queue.job.function">
-        <field name="model_id" ref="model_edi_backend" />
+    <record id="job_fun_exchange_record_generate" model="queue.job.function">
+        <field name="model_id" ref="model_edi_exchange_record" />
         <field name="method">exchange_generate</field>
```
Comment (etobella) [inline edi_oca/data/job_function.xml]: `action_exchange_generate`, isn't it?

Comment (simahawk) [reply edi_oca/data/job_function.xml]: good catch Quique! Tnx :)

**Comment by simahawk** [comment]
test failure seems unrelated
```
2022-03-17 08:08:18,090 400 ERROR odoo odoo.addons.account_invoice_import_invoice2data.tests.test_invoice_import: ERROR: TestInvoiceImport.test_import_free_invoice
Traceback (most recent call last):
  File "/__w/edi/edi/account_invoice_import_invoice2data/wizard/account_invoice_import.py", line 53, in invoice2data_parse_invoice
    invoice2data_res = extract_data(file_name, templates=templates)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/main.py", line 93, in extract_data
    return t.extract(optimized_str)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 204, in extract
    result = parsers.regex.parse(self, {"regex": v, "type": "date"}, optimized_str, True)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/parsers/regex.py", line 36, in parse
    result[k] = template.coerce_type(v, settings["type"])
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 151, in coerce_type
    return self.parse_date(value)
  File "/opt/odoo-venv/lib/python3.6/site-packages/invoice2data/extract/invoice_template.py", line 136, in parse_date
    languages=self.options["languages"],
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/conf.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/__init__.py", line 61, in parse
    data = parser.get_date_data(date_string, date_formats)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 429, in get_date_data
    locale, date_string, date_formats, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 178, in parse
    return instance._parse()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 182, in _parse
    date_data = self._parsers[parser_name]()
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 196, in _try_freshness_parser
    return freshness_date_parser.get_date_data(self._get_translated_date(), self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/date.py", line 235, in _get_translated_date
    self.date_string, keep_formatting=False, settings=self._settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 131, in translate
    relative_translations = self._get_relative_translations(settings=settings)
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 158, in _get_relative_translations
    self._generate_relative_translations(normalize=True))
  File "/opt/odoo-venv/lib/python3.6/site-packages/dateparser/languages/locale.py", line 172, in _generate_relative_translations
    pattern = DIGIT_GROUP_PATTERN.sub(r'?P<n>\d+', pattern)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/regex.py", line 700, in _compile_replacement_helper
    is_group, items = _compile_replacement(source, pattern, is_unicode)
  File "/opt/odoo-venv/lib/python3.6/site-packages/regex/_regex_core.py", line 1736, in _compile_replacement
    raise error("bad escape \\%s" % ch, source.string, source.pos)
regex._regex_core.error: bad escape \d at position 7
```

File: edi_oca/models/edi_exchange_record.py [hunk 2]
```diff
@@ -110,11 +110,11 @@ class EDIExchangeRecord(models.Model):
         ),
     ]
 
-    @api.depends("type_id.code", "model", "res_id")
-    def _compute_name(self):
+    @api.depends("model", "res_id")
```
Comment (mmequignon) [inline edi_oca/models/edi_exchange_record.py:113]: this depends looks wrong?
```python
@api.depends("record", "record.display_name")
```

File: edi_oca/templates/exchange_chatter_msg.xml [hunk 3]
```diff
@@ -13,7 +13,7 @@
                     t-att-data-oe-model="exchange_record._name"
                     t-att-data-oe-id="exchange_record.id"
                 >
-                    <span t-field="exchange_record.name" />
+                    <span t-field="exchange_record.identifier" />
```
Comment (mmequignon) [inline edi_oca/templates/exchange_chatter_msg.xml:16]: why identifier ? I was expecting `related_name`

**Review by mmequignon — APPROVED** [review]
LG, just tiny questions

File: edi_oca/models/edi_exchange_record.py (hunk 2 above)
Comment (simahawk) [reply edi_oca/models/edi_exchange_record.py:113]: is not :) `record` is a property computed from res_id and model ;)

File: edi_oca/templates/exchange_chatter_msg.xml (hunk 3 above)
Comment (simahawk) [reply edi_oca/templates/exchange_chatter_msg.xml:16]: this msg is posted on the related record to say that it has been created or update BY the exc record

**Comment by simahawk** [comment]
Build fixed here https://github.com/OCA/edi/pull/547

**Comment by etobella** [comment]
@simahawk removal of some functions is giving problems to other areas :sob: https://github.com/OCA/l10n-spain/runs/5657494643?check_suite_focus=true

**Comment by simahawk** [comment]
@etobella ah because you are using it directly? I see, well, we can restore them and deprecate them... sorry, I didn't know :sweat_smile: .

What do you prefer? Adapt your implementation or adding back and depracating?

**Comment by simahawk** [comment]
I'd prefer "adapting" but just because it's cleaner :stuck_out_tongue_winking_eye:

**Comment by etobella** [comment]
I would prefer adding them, because it happens to me, but it could happen somewhere else. Just marking them as deprecated and removal on 16 :wink:

**Pull request merged**